### PR TITLE
Added network reducer and redux-react framework

### PR DESCRIPTION
**Pull request by davidjyee**
Added basic redux-react network reducer, is viewable from browser extension (installation instructions have been added to read me). 

Will be adding initialization and working through implementation of Host on redux next to ensure compatibility, just wanted to merge current framework to dev so we are all working on the same framework together.

### CODE REVIEW

File: src/components/Home/Home.tsx [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
+import { Grid, GridItem, Heading, Divider } from '@chakra-ui/react';
+
 export default function Home() {
   return (
-    <div className="HomePage">
-      <p> This is a test paragraph. </p>
-    </div>
+    <Grid w="100%" p={4} gap={4}>
+      <Heading> Home </Heading>
+    </Grid>
```
Comment (dkhl65) [inline src/components/Home/Home.tsx:9]: Please update Home.test.tsx so it passes.

**Review by dkhl65 — CHANGES_REQUESTED** [review]
Make sure unit tests are passing before merge.

File: src/components/Home/Home.tsx (hunk 1 above)
Comment (davidjyee) [reply src/components/Home/Home.tsx:9]: Fixed

**Review by dkhl65 — APPROVED** [review]

File: src/components/Home/Home.test.tsx [hunk 2]
```diff
@@ -2,8 +2,6 @@ import React from 'react';
 import { render, screen } from '@testing-library/react';
 import Home from '../Home';
 
-test('renders test paragraph for homepage', () => {
+test('home is rendered', () => {
   render(<Home />);
```
Comment (ChenyanWang) [inline src/components/Home/Home.test.tsx:6]: Update test to check for "Home" header

File: src/components/Network/Network.tsx [hunk 3]
```diff
@@ -20,5 +20,125 @@ import NetworkObj from '../../services/Network';
 // }
 
 export default function Network() {
-  return <div className="NetworkView">Network View</div>;
+  return <div> Network View </div>;
+
+  // let network = new NetworkObj(5);
```
Comment (ChenyanWang) [inline src/components/Network/Network.tsx:25]: Remove unused comments

File: src/reducers/network.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      const stepAmount = action.payload;
+
+      state.timeCounter += stepAmount;
+      for (let i = 0; i < state.nodeList.length; i++) {
+        state.nodeList[i].getAction(state.timeCounter); // TODO: Only retrieves action, need to perform it in a way that does not mutate!!
+      }
+    },
+    [initNetwork.type]: (state, action) => {
+      state.timeCounter = 1;
+      // for (let nodeid = 0; nodeid < numNodes; nodeid++){
```
Comment (ChenyanWang) [inline src/reducers/network.ts:25]: Please remove unused code

Comment (ChenyanWang) [reply src/reducers/network.ts:25]: Move this to the network test

**Review by ChenyanWang — CHANGES_REQUESTED** [review]
Please fix the empty "home" page test

Will approve after, as we need react redux for other dev
Note: still need to fix porting of network class (as it is now empty) and remove unused code comments

**Review by ChenyanWang — APPROVED** [review]
LGTM!

**Review by RafsanHaque — APPROVED** [review]

**Pull request merged**